VendingMachineApp/VendingMachineApp/Source/AppDelegate.swift

```diff
@@ -13,60 +13,27 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     
     var window: UIWindow?
     
-    var presenter: VendingMachinePresenterType?
-    
     func application(_ application: UIApplication,
                      didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        load()
+        guard
+            var view = window?.rootViewController as? VendingMachineViewType
+            else { return true }
+        view.presenter = VendingMachinePresenter.shared
```

Comment:
view.presenter = VendingMachinePresenter.shared 
이렇게 싱글톤 객체를 외부에서 넣어주면, 
백그라운드에 들어갔다가 나올 때마다 `view.present` 와 `VendingMachinePresenter.shared` 달라지지 않나요?
자판기 객체가 계속 바뀌는 현상이 생기지 않는지 확인해보세요.

Reply:
싱글턴으로 되어있어서 객체가 변경되지는 않는것같습니다. 혹시 어떤 경우에 변경이 될수 있을 까요?
```
Foreground: Optional(<VendingMachineApp.VendingMachinePresenter: 0x600001099c70>)
2019-10-21 17:28:11.649114+0900 VendingMachineApp[31062:14549899] Can't end BackgroundTask: no background task exists with identifier 5 (0x5), or it may have already been ended. Break in UIApplicationEndBackgroundTaskError() to debug.
Foreground: Optional(<VendingMachineApp.VendingMachinePresenter: 0x600001099c70>)
2019-10-21 17:28:14.584907+0900 VendingMachineApp[31062:14549899] Can't end BackgroundTask: no background task exists with identifier 9 (0x9), or it may have already been ended. Break in UIApplicationEndBackgroundTaskError() to debug.
Foreground: Optional(<VendingMachineApp.VendingMachinePresenter: 0x600001099c70>)
```

Reply:
객체가 안 바뀌면 아카이브했던 데이터가 불러와도 안 바뀌는건가요?

Reply:
README.md 도 읽어주세요~>~